plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -647,31 +648,49 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
         else {
             schemaTableNames = ImmutableList.of(prefix.toSchemaTableName());
         }
-        return schemaTableNames.stream()
-                .flatMap(tableName -> {
-                    try {
+
+        return Lists.partition(schemaTableNames, GET_METADATA_BATCH_SIZE).stream()
+                .map(tableBatch -> {
+                    ImmutableList.Builder<TableColumnsMetadata> tableMetadatas = ImmutableList.builderWithExpectedSize(tableBatch.size());
+                    Set<SchemaTableName> remainingTables = new HashSet<>(tableBatch.size());
+                    for (SchemaTableName tableName : tableBatch) {
                         if (redirectTable(session, tableName).isPresent()) {
-                            return Stream.of(TableColumnsMetadata.forRedirectedTable(tableName));
+                            tableMetadatas.add(TableColumnsMetadata.forRedirectedTable(tableName));
+                        }
+                        else {
+                            remainingTables.add(tableName);
                         }
-
-                        Table icebergTable = catalog.loadTable(session, tableName);
-                        List<ColumnMetadata> columns = getColumnMetadatas(icebergTable.schema());
-                        return Stream.of(TableColumnsMetadata.forTable(tableName, columns));
-                    }
-                    catch (TableNotFoundException e) {
-                        // Table disappeared during listing operation
-                        return Stream.empty();
-                    }
-                    catch (UnknownTableTypeException e) {
-                        // Skip unsupported table type in case that the table redirects are not enabled
-                        return Stream.empty();
                     }
-                    catch (RuntimeException e) {
-                        // Table can be being removed and this may cause all sorts of exceptions. Log, because we're catching broadly.
-                        log.warn(e, "Failed to access metadata of table %s during streaming table columns for %s", tableName, prefix);
-                        return Stream.empty();
+
+                    Map<SchemaTableName, List<ColumnMetadata>> loaded = catalog.tryGetColumnMetadata(session, ImmutableList.copyOf(remainingTables));
+                    loaded.forEach((tableName, columns) -> {
+                        remainingTables.remove(tableName);
+                        tableMetadatas.add(TableColumnsMetadata.forTable(tableName, columns));
+                    });
+
+                    for (SchemaTableName tableName : remainingTables) {
+                        try {
+                            Table icebergTable = catalog.loadTable(session, tableName);
```

Comment:
I think we can do better here.
We don't need actually here the "entire" table metadata.

Let's take the schema from the metadata location and load therefore only `1` metadata file instead of many

Reply:
I think this loads 1 metadata file at this point, snapshots and manifests aren't read yet.
I will update `io.trino.plugin.iceberg.TestIcebergMetadataFileOperations#testInformationSchemaColumns` to show that

Reply:
https://github.com/trinodb/trino/pull/18372